### ISSUE TITLE
Including local level terms of bsts in predictions 

### ISSUE THREAD

**Issue by mrojala** [paul-buerkner/brms#147]
When using autocor = cor_bsts(~ ..) as part of the brm model and doing predictions with new data, a warning "Local level terms are currently ignored when 'newdata' is specified." is given. Supporting predictions would be awesome!

**Comment by paul-buerkner** [comment]
I absolutely agree that it would be awesome but it is a lot of work and doesn't fit nicely in the current framework used to predict new data. Essentially, we had to determine when to interpolate to later time points of a time series and then generate predicted local level terms for these time points.

**Comment by paul-buerkner** [comment]
A similar problem applies to ARMA correlation structures which do not support canonical forecasting as well (see https://groups.google.com/forum/#!topic/brms-users/fJqfb8RwykA for a discussion).

**Comment by paul-buerkner** [comment]
A little bit of progress: Predictions for new data in models with ARMA correlation structures should now work.

**Comment by jmorten** [comment]
Would you mind sharing a minimal reprex? Thank you.

**Comment by paul-buerkner** [comment]
Here we go:

```R
# simulate data
y <- arima.sim(list(ar = c(0.7, 0.5, -0.3), ma = -0.3), 300)
dat <- data.frame(y = as.numeric(y))

# fit the model
library(brms)
fit <- brm(y~1, dat, autocor = cor_arma(~1, 3, 1),
           chains = 2, cores = 2)
summary(fit)

# forcast new data
newdata <- rbind(fit$data, data.frame(y = rep(NA, 50)))
pred <- as.data.frame(predict(fit, newdata, nsamples = 1))
names(pred) <- c("est", "se", "lower", "upper")
pred$y <- newdata$y
pred$x <- seq_len(nrow(newdata))
ggplot(pred, aes(x, est, ymin = lower, ymax = upper)) + 
  geom_smooth(stat = "identity") + 
  geom_point(aes(x, y), inherit.aes = FALSE)
```

I chose `nsamples = 1` to better demonstrate how the ARMA process continuous beyond the range of observed data.

**Comment by jmorten** [comment]
What implications does "Warning message:
Local level terms are currently ignored when 'newdata' is specified. " on the predictions?

**Comment by paul-buerkner** [comment]
This basically says that the `cor_bsts` structure is not yet included in the predictions at all, which is the reason this issue was opened in the first place. The progress I reported above, was just related to predictions for the `cor_arma` structure. Implementing predictions with the `cor_bsts` is conceptually not trivial, since this structure models parameters as belonging to specific observations and thus prediction of new data is only possible in relation to the original data. Accordingly, to make sure such predictions are reasonable, we need an interface in which the user tells how the new data relate to the original data.

**Comment by jmorten** [comment]
Thank you.

**Comment by paul-buerkner** [comment]
Closed for reasons explained in #148.
